### PR TITLE
For lima: point /var/run/docker.sock to the lima-managed docker socket file

### PR DESCRIPTION
**Pull request by ericpromislow**
Do this check only when the lima backend is running,
and then only do that if we're running a Moby container engine,
and the container engine just changed.

Signed-off-by: Eric Promislow <epromislow@suse.com>

### CODE REVIEW

File: src/k8s-engine/lima.ts [hunk 1]
```diff
@@ -1164,4 +1164,58 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
...
+set -ex
+
+ln -sf "${ targetPath }" "${ symlinkPath }"
+# chown root:daemon "${ symlinkPath }"
+chmod au+rx "${ symlinkPath }"
+`,
+    { mode: 0o700 });
+    try {
+      const x = await new Promise((resolve, reject) => {
+        sudo.exec(tmpScript, { name: 'Rancher Desktop' }, (error, stdout, stderr) => {
```
Comment (jandubois) [inline src/k8s-engine/lima.ts:1199]: Is this going to be tied into the combined `sudo` handling once the vmnet changes have been merged?

Comment (ericpromislow) [reply src/k8s-engine/lima.ts:1199]: Good point, and I thought of that, although this is a different implementation.

The difference is that the vmnet changes always need to be done (usually on a first run).  These changes don't need to be done if the user never runs moby. So I decided to leave the (future merged code) making two separate calls to `sudo.exec`, and test to verify that it only prompts the user for the password the first time.  If there are two prompts, then this code should be moved into the installation code from the vmnet PR, to keep things simple.

So this PR depends on PR 883.

**Comment by ericpromislow** [comment]
Converting to draft status until PR 883 is merged.

**Review by jandubois — APPROVED** [review]
LGTM (and works).

**Pull request merged**